### ISSUE TITLE
Migrate from catch to doctest?

### ISSUE THREAD

**Issue by bjodah** [symengine/symengine#2028]
I have successfully been able to reduce build times in some projects of mine by switching from catch to [doctest](https://github.com/doctest/doctest) (see graph in linked README to get an idea of the magnitude of build time reduction).

If you are open to making the switch I can open a PR, in which any speedups should already be evident (maybe it will take two commits so that ccache artifacts get used).

**Comment by certik** [comment]
Yes, definitely. I've been using doctest for years, it's great. The speed of compilation is essential.

Yes, go ahead and submit a PR. Thanks!
